### PR TITLE
Height of binary search tree added

### LINKED ISSUE #334: Article on Height of Binary Search Tree
## 🚀 Feature

I would like to add an article for finding height of BST

### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#reporting-new-issues)?

(Yes)

### PR DESCRIPTION
**Pull request by shivammahajancse**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

(Yes)

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

Fixes #334

### CODE REVIEW

**Comment by shivammahajancse** [comment]
Please review @ankitaggarwal23 @siddharth25pandey @atarax665 sir

File: docs/data-structures/height_BST.mdx [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+
+ In the above example, we are given a binary search tree(BST).
+ If the tree is empty then simply return height=0;
+ Otherwise, calculate the maximum of height of left child and height right child.
+ Simply add 1(for root node) after each recursive call for calculating height of BST. 
+
+```
+## Algorithm
+
+For calculating Height of BST, we follow the following algorithm:
```
Comment (siddharth25pandey) [inline docs/data-structures/height_BST.mdx:28]: Do write "following **steps**"

File: docs/data-structures/height_BST.mdx [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+  defaultValue="cpp"
+  values={[
+    { label: 'C++', value: 'cpp', },
+    { label: 'Python', value: 'py', },
+  ]
+}>
+<TabItem value="cpp">
+
+```cpp
+#include<iostream>
```
Comment (siddharth25pandey) [inline docs/data-structures/height_BST.mdx:53]: Use **bits/stdc+**

File: docs/data-structures/height_BST.mdx [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+<Tabs
+  defaultValue="cpp"
+  values={[
+    { label: 'C++', value: 'cpp', },
+    { label: 'Python', value: 'py', },
+  ]
+}>
+<TabItem value="cpp">
+
+```cpp
```
Comment (siddharth25pandey) [inline docs/data-structures/height_BST.mdx:52]: Write C++ instead of Cpp

File: docs/data-structures/height_BST.mdx [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+    {
+      return max(heightofbst(root->left),heightofbst(root->right))+1;
+    }
+    
+}
+//Main Function began
+int main() {
+    Node* root = NULL;
+    //input values
+     int n;
```
Comment (siddharth25pandey) [inline docs/data-structures/height_BST.mdx:100]: Write good naming convention

File: docs/data-structures/height_BST.mdx [hunk 5]
```diff
@@ -0,0 +1,202 @@
...
+     cout<<"Enter the size of tree: ";
+     cin>>n;
+    while(n--){
+        int d;
+        cout<<"Enter the data: ";
+        cin>>d;
```
Comment (siddharth25pandey) [inline docs/data-structures/height_BST.mdx:106]: Write good naming convention

**Review by siddharth25pandey — CHANGES_REQUESTED** [review]
@shivammahajancse Do the suggested changes

File: docs/data-structures/height_BST.mdx (hunk 1 above)
Comment (shivammahajancse) [reply docs/data-structures/height_BST.mdx]: done

File: docs/data-structures/height_BST.mdx (hunk 3 above)
Comment (shivammahajancse) [reply docs/data-structures/height_BST.mdx:52]: done

File: docs/data-structures/height_BST.mdx (hunk 4 above)
Comment (shivammahajancse) [reply docs/data-structures/height_BST.mdx]: done

File: docs/data-structures/height_BST.mdx (hunk 5 above)
Comment (shivammahajancse) [reply docs/data-structures/height_BST.mdx]: done

**Comment by shivammahajancse** [comment]
@siddharth25pandey merge conflict solved

**Comment by shivammahajancse** [comment]
@siddharth25pandey conflict solved

**Review by siddharth25pandey — APPROVED** [review]
@shivammahajancse LGTM 💯 Good to go

**Pull request merged**